presto-spark/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -0,0 +1,345 @@
+            JsonCodec<PrestoSparkTaskDescriptor> taskDescriptorJsonCodec,
+            JsonCodec<TaskStats> taskStatsJsonCodec,
+            Executor notificationExecutor,
+            ScheduledExecutorService yieldExecutor,
+            LocalExecutionPlanner localExecutionPlanner,
+            BlockEncodingSerde blockEncodingSerde,
+            DataSize maxUserMemory,
+            DataSize maxTotalMemory,
+            DataSize maxSpillMemory,
+            boolean perOperatorCpuTimerEnabled,
+            boolean cpuTimerEnabled)
+    {
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+        this.taskDescriptorJsonCodec = requireNonNull(taskDescriptorJsonCodec, "sparkTaskDescriptorJsonCodec is null");
+        this.taskStatsJsonCodec = requireNonNull(taskStatsJsonCodec, "taskStatsJsonCodec is null");
+        this.notificationExecutor = requireNonNull(notificationExecutor, "notificationExecutor is null");
+        this.yieldExecutor = requireNonNull(yieldExecutor, "yieldExecutor is null");
+        this.localExecutionPlanner = requireNonNull(localExecutionPlanner, "localExecutionPlanner is null");
+        this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
+        this.maxUserMemory = requireNonNull(maxUserMemory, "maxUserMemory is null");
+        this.maxTotalMemory = requireNonNull(maxTotalMemory, "maxTotalMemory is null");
+        this.maxSpillMemory = requireNonNull(maxSpillMemory, "maxSpillMemory is null");
+        this.perOperatorCpuTimerEnabled = perOperatorCpuTimerEnabled;
+        this.cpuTimerEnabled = cpuTimerEnabled;
+    }
+
+    @Override
+    public IPrestoSparkTaskExecutor create(
+            int partitionId,
+            int attemptNumber,
+            SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor,
+            PrestoSparkTaskInputs inputs,
+            CollectionAccumulator<SerializedTaskStats> taskStatsCollector)
+    {
+        PrestoSparkTaskDescriptor taskDescriptor = taskDescriptorJsonCodec.fromJson(serializedTaskDescriptor.getBytes());
+
+        Session session = taskDescriptor.getSession().toSession(sessionPropertyManager, taskDescriptor.getExtraCredentials());
+        PlanFragment fragment = taskDescriptor.getFragment();
+        StageId stageId = new StageId(session.getQueryId(), fragment.getId().getId());
+        // TODO: include attemptId in taskId
+        TaskId taskId = new TaskId(new StageExecutionId(stageId, 0), partitionId);
+
+        MemoryPool memoryPool = new MemoryPool(new MemoryPoolId("spark-executor-memory-pool"), maxTotalMemory);
+        SpillSpaceTracker spillSpaceTracker = new SpillSpaceTracker(maxSpillMemory);
+
+        QueryContext queryContext = new QueryContext(
+                session.getQueryId(),
+                maxUserMemory,
+                maxTotalMemory,
+                memoryPool,
+                new TestingGcMonitor(),
```

Comment:
https://github.com/prestodb/presto/pull/13760#discussion_r366092923

I'm not sure if it makes sense to create a new class. Maybe we can simply rename the `TestingGcMonitor` to something like `CountingGcMonitor`?

Anyhow it can be done as a separate PR.